webapp-django/crashstats/manage/forms.py

```diff
@@ -40,6 +42,84 @@ class FilterEventsForm(BaseForm):
     action = forms.CharField(required=False)
 
 
+class FilterAPITokensForm(BaseForm):
+
+    user = forms.CharField(required=False)
+    key = forms.CharField(required=False)
+    expired = forms.CharField(required=False)
+
+    def clean_expired(self):
+        value = self.cleaned_data['expired']
+        if value not in ('yes', 'no', ''):
+            raise forms.ValidationError('Invalid value')
+        return value
+
+
+class APITokenForm(BaseModelForm):
+
+    user = forms.CharField(required=True)
+    expires = forms.ChoiceField(required=True)
+
+    class Meta:
+        model = Token
+        fields = ('user', 'expires', 'notes', 'permissions')
+
+    def __init__(self, *args, **kwargs):
+        self.possible_permissions = kwargs.pop('possible_permissions', [])
+        expires_choices = kwargs.pop('expires_choices', [])
+        super(APITokenForm, self).__init__(*args, **kwargs)
+        self.fields['permissions'].choices = (
+            (x.pk, x.name) for x in self.possible_permissions
+        )
+        self.fields['user'].widget = forms.widgets.TextInput()
+        self.fields['user'].label = 'User (by email)'
+        self.fields['expires'].choices = expires_choices
+
+    def clean_user(self):
+        value = self.cleaned_data['user']
+        try:
+            user = User.objects.get(email__istartswith=value.strip())
```

Comment:
@peterbe You didn't address this nor explained why you didn't. Should I be worried? :-)